src/shared/components/palette/collection/collection-create-checker.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { CaseField } from '../../../domain/definition';
+import { Profile } from '../../../domain/profile';
+
+@Injectable()
+export class CollectionCreateCheckerService {
+
+  public setDisplayContextForChildren(caseField: CaseField, caseFields: CaseField[], profile: Profile) {
```

Comment:
Shouldn't this be recursive? At the moment it only updates one level below.

Reply:
right, added 👍